app/org/sagebionetworks/bridge/services/HealthCodeServiceImpl.java

```diff
@@ -43,15 +44,43 @@ public String getCode() {
     }
 
     @Override
-    public String getHealthCode(String id) {
-        return healthIdDao.getCode(id);
+    public HealthId getMapping(String healthId) {
+        checkNotNull(healthId);
+        final String healthCode = healthIdDao.getCode(healthId);
+        if (healthCode == null) {
+            return null;
+        }
+        return getHealthIdObject(healthId, healthCode);
     }
-
+    
     @Override
-    public String getStudyIdentifier(String healthCode) {
-        return healthCodeDao.getStudyIdentifier(healthCode);
+    public HealthId getMapping(Account account) {
+        checkNotNull(account);
+        
+        final String healthId = account.getHealthId();
+        if (healthId == null) {
+            return null;
+        }
+        final String healthCode = healthIdDao.getCode(healthId);
+        if (healthCode == null) {
+            return null;
+        }
+        return getHealthIdObject(healthId, healthCode);
     }
 
+    private HealthId getHealthIdObject(final String healthId, final String healthCode) {
+        return new HealthId() {
+            @Override
+            public String getId() {
+                return healthId;
+            }
+            @Override
+            public String getCode() {
+                return healthCode;
+            }
+        };
```

Comment:
[POLISH] Instead of an anonymous local class, it would be much cleaner to make this a full-fledged class (similar to the StudyIdentifier class).